sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureSeekableByteChannel.java

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.nio;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.NonReadableChannelException;
+import java.nio.channels.NonWritableChannelException;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+
+/**
+ * A byte channel that maintains a current position.
+ * <p>
+ * This type is primarily offered to support some jdk convenience methods such as
+ * {@link Files#createFile(Path, FileAttribute[])} which requires opening a channel and closing it. A channel may only
+ * be opened in read mode OR write mode. It may not be opened in read/write mode. Seeking is supported for reads, but
+ * not for writes. Modifications to existing files is not permitted--only creating new files or overwriting existing
+ * files.
+ * <p>
+ * This type is not threadsafe to prevent having to hold locks across network calls.
+ * <p>
+ * {@link NioBlobInputStream} and {@link NioBlobOutputStream} are the preferred types for reading and writing blob data
+ * and are used internally by this type.
+ */
+public class AzureSeekableByteChannel implements SeekableByteChannel {
```

Comment:
Should move this class into `implementation` as this is an implementation detail? At minimum we should make this class final.

Reply:
Good call on the final. I wasn't going to put it in implementation because our other types like NioBlobInputStream aren't in any special implementation package. We actually have an issue open asking us to make a method on AzurePath public, so I think it's a scenario where sometimes customers will want to get more provider specific functionality out of the types, so the same could apply here. 

I'm open to the conversation if we do want to move those types into implementation, though.